drivers/storage/portworx/components.go

```diff
@@ -110,55 +110,66 @@ func (p *portworx) installComponents(cluster *corev1alpha1.StorageCluster) error
 	if err = p.setupPortworxService(t); err != nil {
 		return err
 	}
+
 	if err = p.setupPortworxAPI(t); err != nil {
-		return err
+		msg := fmt.Sprintf("Failed to setup Portworx API. %v", err)
+		p.warningEvent(cluster, util.FailedComponentReason, msg)
 	}
+
 	if err = p.createPortworxStorageClasses(t); err != nil {
 		msg := fmt.Sprintf("Failed to create default Portworx storage classes. %v", err)
-		p.warningEvent(cluster, failedComponentReason, msg)
-		return nil
+		p.warningEvent(cluster, util.FailedComponentReason, msg)
 	}
+
 	if err = p.createCustomResourceDefinitions(); err != nil {
```

Comment:
I was thinking that if we fail to create the CRD then we should not start Portworx anyway as some of it's functionality may not work. It would be bad to debug that later, instead just throw an error upfront. That's why I am returning an error here.